### PR TITLE
fix(swap): change limit when pool change

### PR DESCRIPTION
**Pull request by banklesss**
Related to [YOMO-857](https://emurgo.atlassian.net/browse/YOMO-857)

[YOMO-857]: https://emurgo.atlassian.net/browse/YOMO-857?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by stackchain** [comment]
I believe this should happen at the state level.

**Comment by banklesss** [comment]
> I believe this should happen at the state level.

i will take a look

**Review by jorbuedo — CHANGES_REQUESTED** [review]
Should be a 1 line change inside
```
      case SwapCreateOrderActionType.SelectedPoolChanged:
```

**Comment by banklesss** [comment]
Working on it already

```
case SwapCreateOrderActionType.SelectedPoolChanged:
        if (state.orderData.type === 'market') break

        draft.orderData.selectedPoolId = action.poolId

        draft.orderData.selectedPoolCalculation =
          selectedPoolCalculationSelector(draft.orderData)

        if (draft.orderData.selectedPoolCalculation === undefined) break

        draft.orderData.amounts.buy =
          draft.orderData.selectedPoolCalculation.sides.buy

        draft.orderData.limitPrice = getMarketPrice(
          draft.orderData.selectedPoolCalculation.pool,
          draft.orderData.selectedPoolCalculation.sides.sell,
        )
        break

```

**Review by michalszorad — APPROVED** [review]

File: packages/swap/src/translators/reactjs/state/state.ts [hunk 1]
```diff
@@ -312,6 +313,11 @@ const orderReducer = (
 
         draft.orderData.amounts.buy =
           draft.orderData.selectedPoolCalculation.sides.buy
+
+        draft.orderData.limitPrice = getMarketPrice(
+          draft.orderData.selectedPoolCalculation.pool,
+          draft.orderData.selectedPoolCalculation.sides.sell,
+        )
```
Comment (stackchain) [inline packages/swap/src/translators/reactjs/state/state.ts]:
```suggestion
        draft.orderData.limitPrice = draft.orderData.bestPoolCalculation.prices.market
```

Comment (stackchain) [reply packages/swap/src/translators/reactjs/state/state.ts]: it's ready in the calculation obj.

File: packages/swap/src/translators/reactjs/state/state.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import {SwapDiscountTier} from '../../../translators/constants'
 import {makeOrderCalculations} from '../../../helpers/orders/factories/makeOrderCalculations'
 import {selectedPoolCalculationSelector} from './selectors/selectedPoolCalculationSelector'
 import {getBestPoolCalculation} from '../../../helpers/pools/getBestPoolCalculation'
+import {getMarketPrice} from '../../../helpers/prices/getMarketPrice'
```
Comment (stackchain) [inline packages/swap/src/translators/reactjs/state/state.ts]:
```suggestion
```

File: packages/swap/src/translators/reactjs/state/state.ts [hunk 3]
```diff
@@ -312,6 +312,8 @@ const orderReducer = (
 
         draft.orderData.amounts.buy =
           draft.orderData.selectedPoolCalculation.sides.buy
+
+        draft.orderData.limitPrice = draft.orderData.bestPoolCalculation.prices.market
```
Comment (stackchain) [inline packages/swap/src/translators/reactjs/state/state.ts]:
```suggestion
        draft.orderData.limitPrice = draft.orderData.selectedPoolCalculation.prices.market
```

Comment (stackchain) [reply packages/swap/src/translators/reactjs/state/state.ts]: meh

File: packages/swap/src/translators/reactjs/state/state.ts [hunk 4]
```diff
@@ -312,6 +312,8 @@ const orderReducer = (
 
         draft.orderData.amounts.buy =
           draft.orderData.selectedPoolCalculation.sides.buy
+
+        draft.orderData.limitPrice = draft.orderData.selectedPoolCalculation.prices.market
```
Comment (stackchain) [inline packages/swap/src/translators/reactjs/state/state.ts]:
```suggestion
        draft.orderData.limitPrice =
          draft.orderData.selectedPoolCalculation.prices.market
```

**Review by jorbuedo — APPROVED** [review]

**Pull request merged**